internal/cli/login/aap.go

```diff
@@ -31,33 +34,48 @@ func (c *AAPRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	return resp, nil
 }
 
-func NewAAPOAuth2Config(caFile, clientId, authUrl string, insecure bool) AAPOAuth {
-	return AAPOAuth{
+func NewAAPOAuth2Config(metadata api.ObjectMeta, spec api.AapProviderSpec, clientId, caFile string, insecure bool, apiServerURL string) *AAPOAuth {
+	return &AAPOAuth{
+		Metadata:           metadata,
+		Spec:               spec,
+		ClientId:           clientId,
 		CAFile:             caFile,
 		InsecureSkipVerify: insecure,
-		ClientId:           clientId,
-		ConfigUrl:          authUrl,
+		ApiServerURL:       apiServerURL,
 	}
 }
 
-func (o AAPOAuth) getOAuth2Config() OauthServerResponse {
-	return OauthServerResponse{
-		TokenEndpoint: fmt.Sprintf("%s/o/token/", o.ConfigUrl),
-		AuthEndpoint:  fmt.Sprintf("%s/o/authorize/", o.ConfigUrl),
-	}
+func (o *AAPOAuth) SetInsecureSkipVerify(insecureSkipVerify bool) {
+	o.InsecureSkipVerify = insecureSkipVerify
 }
 
-func (o AAPOAuth) getOAuth2Client(callback string) (*osincli.Client, error) {
-	oauthServerResponse := o.getOAuth2Config()
+func (o *AAPOAuth) getOAuth2Client(callback string) (*osincli.Client, error) {
+	authUrl := o.Spec.ApiUrl
+	if o.Spec.ExternalApiUrl != nil {
+		authUrl = *o.Spec.ExternalApiUrl
+	}
+
+	codeVerifier, codeChallenge, err := generatePKCEVerifier()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate PKCE parameters: %w", err)
+	}
+
+	// Use the API server's token proxy endpoint instead of the AAP provider's token endpoint
+	if o.Metadata.Name == nil {
+		return nil, fmt.Errorf("provider name is required")
+	}
 
 	config := &osincli.ClientConfig{
 		ClientId:                 o.ClientId,
-		AuthorizeUrl:             oauthServerResponse.AuthEndpoint,
-		TokenUrl:                 oauthServerResponse.TokenEndpoint,
+		AuthorizeUrl:             fmt.Sprintf("%s/o/authorize/", authUrl),
```

Comment:
Is this format the only acceptable format ?
Why is the URL (in the spec) not in its final format ?